lib/bamboo/adapters/send_grid_helper.ex

```diff
@@ -294,4 +295,25 @@ defmodule Bamboo.SendGridHelper do
     email
     |> Email.put_private(@ip_pool_name_field, ip_pool_name)
   end
+
+  @doc """
+  A map of unique arguments for this email. This will override any existing unique arguments.
+
+  ## Example
+
+      email
+      |> with_unique_args(%{new_arg_1: "new arg 1", new_arg_2: "new arg 2"})
+  """
+  def with_unique_args(email, unique_args) when is_map(unique_args) do
+    unique_args =
+      Map.get(email.private, @unique_args, %{})
```

Comment:
What do you think about using [`Map.update/4`](https://hexdocs.pm/elixir/1.12/Map.html#update/4) here instead of getting the values and then updating them? I think it reveals intention slightly more clearly.

Reply:
I tried using `Map.update/4` but I had difficulty on converting my current implementation. 
```
Map.update(email.private, @unique_args, %{}, &Map.merge(&1, unique_args))
```
`Map.update/4` returns not only the values but also the keys and `Email.put_private` expect 3 arguments. Can I ask for your guidance on this one.

Reply:
If you think `Map.update/4` makes it more confusing. I'm okay leaving things as they are. No need to change this. Thanks for looking into it!